funnel/templates/event.html

```diff
@@ -46,37 +46,251 @@
-          <td class='js-searchable'>{{ p.company }}</td>
+    <tbody id="participants-table-content">
+      {% raw %}
+      <script id='participant-row' type='text/ractive'>
+      {{#each participants}}
+        <tr id="p-{{ pid }}">
+          <td class='js-searchable'>{{ fullname }}</td>
+          <td class='js-searchable'>{{ email }}</td>
+          <td class='js-searchable'>{{ company }}</td>
           <td>
             <ul class="list-inline status-btn-list">
             <li>
-              <a href={{ "{0}participant/{1}/badge".format(space.url_for(), p.id) }} class={% if p.badge_printed %} 'btn btn-xs btn-success'  {% else %} 'btn btn-xs btn-info' {% endif %} target="_blank">Show badge</a>
+              <a href="{{ showbadge_url }}" {{#badge_printed}} class="btn btn-xs btn-success" {{/badge_printed}} {{^badge_printed}} class="btn btn-xs btn-info" {{/badge_printed}} target="_blank">Show badge</a>
             </li>
             <li>
-              <a href={{ "{0}participant/{1}/edit".format(space.url_for(), p.id) }} class='btn btn-xs btn-default'>Edit</a>
+              <a href="{{ edit_url }}" class="btn btn-xs btn-default" target="_blank">Edit</a>
             </li>
             <li>
-            {%- if p.checked_in %}
-              <a href="{{ event.name }}/checkin/{{ p.id }}" class='btn btn-xs btn-danger'>Cancel Check-in</a>
-            {%- else %}
-            <a href="{{ event.name }}/checkin/{{ p.id }}?checkin=t" class='btn btn-xs btn-warning'>Check-in</a>
-            {%- endif %}
+            {{#checked_in}}
+              <a href="{{ checkin_url }}/checkin/{{ pid }}" data-pid="{{ pid }}" class="btn btn-xs js-cancel-checkin">Cancel Check-in</a>
+              <i class="fa fa-spinner fa-pulse loader"></i>
+              <i class="fa fa-close loader js-abort"></i>
+            {{/checked_in}}
+            {{^checked_in}}
+              <a href="{{ checkin_url }}/checkin/{{ pid }}?checkin=t" data-pid="{{ pid }}"sd class="btn btn-xs js-checkin">Check-in</a>
+              <i class="fa fa-spinner fa-pulse loader"></i>
+              <i class="fa fa-close loader js-abort"></i>
+            {{/checked_in}}
             </li>
             </ul>
           </td>
         </tr>
-      {%- endfor %}
+      {{/each}}
+      </script>
+      {% endraw %}
     </tbody>
   </table>
 {% endblock %}
 {% block footerscripts %}
 <script>
+  if (Modernizr.localstorage) {
+    window.Store = {};
+
+    //Reads from LocalStorage (key is 'checkin-queue' or 'cancelcheckin-queue')
+    window.Store.readFromLocalstorage = function(key) {
+      return JSON.parse(localStorage.getItem(key));
+    }
+    //Adds to LocalStorage (key is 'checkin-queue' or 'cancelcheckin-queue')
+    window.Store.addToLocalstorage = function(key, value) {
+      return localStorage.setItem(key, JSON.stringify(value))
+    }
+
+    window.Queue = {};
+
+    //Adds a participant_id to queue
+    window.Queue.enqueue = function(queueName, participant_id) {
```

Comment:
This isn't really a 'Queue' from a computer science perspective, since the `participant_id` is being pushed to the end of the list (queues are FIFO). This seems like a interface that offers list operations for data stored via `Store`. Maybe call it `StoreList`? Again, if I understand correctly, the the inner details in these functions have nothing to do with 'participants' per se? As in these functions can be applied in a different context with a similar problem of saving lists in stores?